### PR TITLE
PLAT-183 service account

### PR DESCRIPTION
**Pull request by itzikYeret**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/terraform-providers/terraform-provider-dome9
 
 go 1.13
 
+replace github.com/dome9/dome9-sdk-go v1.10.0 => ../dome9-sdk-go
```
Comment (chkp-yogevk) [inline go.mod]: remove local replacment

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/terraform-providers/terraform-provider-dome9
...
+
 require (
 	github.com/dome9/dome9-sdk-go v1.10.0
```
Comment (chkp-yogevk) [inline go.mod]: increase to v1.11.0

File: dome9/data_source_dome9_service_account.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	resp, _, err := d9Client.serviceAccounts.Get(id)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(resp.Id)
+	_ = d.Set("name", resp.Name)
+	_ = d.Set("api_key_id", resp.ApiKeyId)
+	_ = d.Set("role_ids", resp.RoleIds)
+	_ = d.Set("date_created", resp.DateCreated.Format("2006-01-02 15:04:05"))
```
Comment (chkp-yogevk) [inline dome9/data_source_dome9_service_account.go:56]: why this specific date? "2006-01-02 15:04:05"

File: dome9/resource_dome9_service_account.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+)
+
+func resourceServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceAccountCreate,
+		Read:   resourceServiceAccountRead,
+		Update: resourceServiceAccountUpdate,
+		Delete: resourceServiceAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```
Comment (chkp-yogevk) [inline dome9/resource_dome9_service_account.go:17]: what does Importer means? @itzikYeret @paveli-d9

**Review by chkp-yogevk — CHANGES_REQUESTED** [review]
looks good, take a look at my comments, thanks.

File: dome9/data_source_dome9_service_account.go (hunk 3 above)
Comment (itzikYeret) [reply dome9/data_source_dome9_service_account.go:56]: this is just format

File: dome9/resource_dome9_service_account.go (hunk 4 above)
Comment (itzikYeret) [reply dome9/resource_dome9_service_account.go:17]: I don't know, I copied this from another resource

File: dome9/resource_dome9_service_account.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+			"api_key_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"api_key_secret": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"role_ids": {
+				Type:     schema.TypeList,
```
Comment (vasy100) [inline dome9/resource_dome9_service_account.go]: Is "role_ids" represent an ordered collection of items? If no not, change it to another type "TypeSet".

**Review by vasy100 — CHANGES_REQUESTED** [review]
Lock's good, let check it on teamcity from branch.

**Pull request merged**